Packs/Jira/Integrations/JiraV2/JiraV2.py

```diff
@@ -680,30 +754,202 @@ def fetch_incidents(query, id_offset, fetch_by_created=None, **_):
     return incidents
 
 
-def get_remote_data_command(id: str, lastUpdate: str) -> GetRemoteDataResponse:
+def get_attachment_data(attachment):
+    """
+    Get attachments content
+    :param attachment: attachment metadata
+    :return: attachment name and content
+    """
+    attachment_url = f"secure{attachment['content'].split('/secure')[-1]}"
+    filename = attachment_url.split("/")[-1]
+    attachments_zip = jira_req(method='GET', resource_url=attachment_url).content
+    return filename, attachments_zip
+
+
+def get_attachments(attachments, incident_modified_date, only_new=True):
+    """
+    Get incident attachments as fileResults objects
+    :param attachments: the issue's attachments
+    :param incident_modified_date: the date the incident was last updated
+    :param only_new: if 'True', getting only attachments that was added after the incident_modified_date
+    :return: a list of fileResults
+    """
+    file_results = []
+    # list of all attachments
+    if attachments:
+        if not only_new:
+            for attachment in attachments:
+                filename, attachments_zip = get_attachment_data(attachment)
+                file_results.append(fileResult(filename=filename, data=attachments_zip))
+        else:
+            for attachment in attachments:
+                attachment_modified_date: datetime = parse(
+                    str(dict_safe_get(attachment, ['created'], "", str))
+                ).replace(tzinfo=pytz.UTC)
+                if incident_modified_date <= attachment_modified_date:
+                    filename, attachments_zip = get_attachment_data(attachment)
+                    file_results.append(fileResult(filename=filename, data=attachments_zip))
+    return file_results
+
+
+def get_comments(comments, incident_modified_date,  only_new=True):
+    """
+    Get issue's comments
+    :param comments: the issue's comments
+    :param incident_modified_date: the date the incident was last updated
+    :param only_new: if 'True', getting only comments that was added after the incident_modified_date
+    :return: a list of comments
+    """
+    if not only_new:
+        return comments
+    else:
+        returned_comments = []
+        for comment in comments:
+            comment_modified_date: datetime = parse(
+                str(dict_safe_get(comment, ['updated'], "", str))
+            ).replace(tzinfo=pytz.UTC)
+            if incident_modified_date <= comment_modified_date:
+                returned_comments.append(comment)
+        return returned_comments
+
+
+def get_incident_entries(issue, incident_modified_date, only_new=True, should_get_comments=True, should_get_attachments=True):
+    """
+    This function get for incident entries
```

Comment:
be more specific. gets the entries from the incident in order to mirror them out? getting the entries from the ticket to minor in?

Reply:
I use this function not only for the mirroring. but I can add the type of entries.